src/Database/Drivers/SqlsrvDriver.php

```diff
@@ -91,15 +91,28 @@ public function formatLike($value, $pos)
 	 */
 	public function applyLimit(& $sql, $limit, $offset)
 	{
-		if ($limit >= 0) {
-			$sql = preg_replace('#^\s*(SELECT|UPDATE|DELETE)#i', '$0 TOP ' . (int) $limit, $sql, 1, $count);
-			if (!$count) {
-				throw new Nette\InvalidArgumentException('SQL query must begin with SELECT, UPDATE or DELETE command.');
+		$info = $this->connection->query("SELECT CAST(SERVERPROPERTY('ProductVersion') AS varchar) AS [ver]")->fetch();
+
+		if ((int) $info->ver < 11) { // 11 == SQL Server 2012
+			if ($limit >= 0) {
+				$sql = preg_replace('#^\s*(SELECT|UPDATE|DELETE)#i', '$0 TOP ' . (int) $limit, $sql, 1, $count);
+				if (!$count) {
+					throw new Nette\InvalidArgumentException('SQL query must begin with SELECT, UPDATE or DELETE command.');
+				}
 			}
-		}
 
-		if ($offset > 0) {
-			throw new Nette\NotSupportedException('Offset is not supported by this database.');
+			if ($offset > 0) {
+				throw new Nette\NotSupportedException('Offset is not supported by this database.');
+			}
+		} else {
+			if (!preg_match('#ORDER\sBY\s(.*)#i', $sql)) {
```

Comment:
Yes, but its message is somewhat hard to read: 
`SQL Error (153): Incorrect syntax near '2' Invalid usage of the option next in the FETCH statement. `
for SQL:
`SELECT * FROM address OFFSET 2 ROWS FETCH NEXT 5 ROWS ONLY`